web/blog/tenzir-for-splunk-users/index.md

```diff
@@ -0,0 +1,316 @@
+exporting the entire database and only then applying the filter.
+
+Why does this single responsibility principle matter? Because its critical for
+*composition*: we can now replace `export` with another data source, like
+[`from`](/operators/sources/from) [`kafka`](/connectors/kafka) and the rest of
+the pipeline stays the same.
+
+:::note charting
+Tenzir doesn't have charting capabilities (yet), so in this and subsequent
+examples, we end with the data, or replace a `chart` query with an aggregation.
+:::
+
+### Top 10 sources by number of connections
+
+Splunk:
+
+```splunk-spl
+index=zeek sourcetype=zeek_conn
+| top id.orig_h
+| head 10
+```
+
+Tenzir:
+
+```splunk-spl
+export
+| where #schema == "zeek.conn"
+| top id.orig_h
+| head 10
+```
+
+Note the similarity. Like SPL, we support
+[`top`](/operators/transformations/top) and
+[`rare`](/operators/transformations/rare) to make Splunk users feel at home.
+
+### Top 10 sources by bytes sent
+
+Splunk:
+
+```splunk-spl
+index=zeek sourcetype=zeek_conn
+| stats values(service) as Services sum(orig_bytes) as B by id.orig_h
+| sort -B
+| head 10
+| eval MB = round(B/1024/1024,2)
+| eval GB = round(MB/1024,2)
+| rename id.orig_h as Source
+| fields Source B MB GB Services
+```
+
+Tenzir:
+
+```splunk-spl
+export
+| where #schema == "zeek.conn"
+| summarize Services=distinct(service), B=sum(orig_bytes) by id.orig_h
+| sort B desc
+| head 10
+| extend MB=round(B/1024/1024,2)
+| extend GB=round(MB/1024,2)
+| put Source=id.orig_h, B, MB, GB, Services
+```
+
+This example is a bit more involved. Let's take it apart.
+
+- We opted for Kusto's syntax of sorting, by appending an `asc` or `desc`
+  qualifier after the field name. `sort -B` translates into `sort B desc`,
+  whereas `sort B` into `sort B asc`.
+
+- SPL's `eval` maps to [`extend`](/operators/transformations/extend).
+
+- The difference between `extend` and [`put`](/operators/transformations/put) is
+  that `extend` keeps all fields as is, whereas `put` reorders fields and
+  performs an explicit projection with the provided fields.
+
+### Bytes transferred over time by service
+
+Splunk:
+
+```splunk-spl
+index=zeek sourcetype="zeek_conn" OR sourcetype="zeek_conn_long"
+| eval orig_megabytes = round(orig_bytes/1024/1024,2)
+| eval resp_megabytes = round(resp_bytes/1024/1024,2)
+| eval orig_gigabytes = round(orig_megabytes/1024,2)
+| eval resp_gigabytes = round(resp_megabytes/1024,2)
+| timechart sum(orig_gigabytes) AS 'Outgoing Gigabytes',sum(resp_gigabytes) AS 'Incoming Gigabytes' by service span=1h
+```
+
+Tenzir:
+
+```splunk-spl
+export
+| where #schema == /zeek\.conn.*/
+| extend orig_megabytes=round(orig_bytes/1024/1024, 2)
+| extend resp_megabytes=round(orig_bytes/1024/1024, 2)
+| extend orig_gigabytes=round(orig_megabytes/1024, 2)
+| extend resp_gigabytes=round(orig_megabytes/1024, 2)
+| summarize 'Outgoing Gigabytes'=sum(orig_gigabytes), 'Incoming Gigabytes'=sum(resp_gigabytes) by service resolution 1h
```

Comment:
The `by`-clause is missing the timestamp.

Reply:
Should this be `by :timestamp, service`?